[Bug]: `@babel/eslint-parser` will still load root babel config even if `requireConfigFile` is set to `false`

### 💻

- [ ] Would you like to work on a fix?

### How are you using Babel?

@babel/eslint-parser

### Input code

``` javascript
import info from './package.json' assert { type: 'json' };
```

### Configuration file name

`.eslintrc.js`

### Configuration

```javascript
module.exports = {
  env: {
    node: true,
    es2022: true,
    mocha: true,
  },
  overrides: [{
    files: ['*.js', '*.mjs'],
  }],
  extends: [
    'airbnb-base',
  ],
  parserOptions: {
    ecmaVersion: 'latest',
    sourceType: 'module',
    requireConfigFile: false,
    babelOptons: {
      plugins: [
        '@babel/plugin-syntax-import-assertions',
      ],
    },
  },
  parser: '@babel/eslint-parser',
  rules: {
  },
};
```

### Current and expected behavior

Not able to enable importAssertions plugin:
```
..\Dev\account-service\index.mjs
  3:34  error  Parsing error: This experimental syntax requires enabling the parser plugin: "importAssertions". (3:34)
```

### Environment

```
  System:
    OS: Windows 10 10.0.19042
  Binaries:
    Node: 17.6.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.17 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 8.5.1 - C:\Program Files\nodejs\npm.CMD
  npmPackages:
    @babel/eslint-parser: ^7.17.0 => 7.17.0
    @babel/plugin-syntax-import-assertions: ^7.16.7 => 7.16.7 
    eslint: ^8.10.0 => 8.10.0
```

### Possible solution

_No response_

### Additional context

_No response_

--- a/eslint/babel-eslint-parser/README.md
+++ b/eslint/babel-eslint-parser/README.md
@@ -61,6 +61,8 @@ module.exports = {
     babelOptions: {
       babelrc: false,
       configFile: false,
+      // your babel options
+      presets: ["@babel/preset-env"],
     },
   },
 };

--- a/eslint/babel-eslint-parser/README.md
+++ b/eslint/babel-eslint-parser/README.md
@@ -49,7 +49,23 @@ With the parser set, your configuration can be configured as described in the [C
 
 Additional configuration options can be set in your ESLint configuration under the `parserOptions` key. Please note that the `ecmaFeatures` config property may still be required for ESLint to work properly with features not in ECMAScript 5 by default.
 
-- `requireConfigFile` (default `true`) can be set to `false` to allow @babel/eslint-parser to run on files that do not have a Babel configuration associated with them. This can be useful for linting files that are not transformed by Babel (such as tooling configuration files), though we recommend using the default parser via [glob-based configuration](https://eslint.org/docs/user-guide/configuring/configuration-files#configuration-based-on-glob-patterns). Note: @babel/eslint-parser will not parse any experimental syntax when no configuration file is found.
+- `requireConfigFile` (default `true`) can be set to `false` to allow @babel/eslint-parser to run on files that do not have a Babel configuration associated with them. This can be useful for linting files that are not transformed by Babel (such as tooling configuration files), though we recommend using the default parser via [glob-based configuration](https://eslint.org/docs/user-guide/configuring/configuration-files#configuration-based-on-glob-patterns).
+  Note: When `requireConfigFile` is `false`, @babel/eslint-parser will still try to load the root babel config. If no configuration file is found, @babel/eslint-parser will not parse any experimental syntax. Though not recommended, if you have a babel config, but would like to prevent @babel/eslint-parser from loading Babel config, please specify
+  **.eslintrc.js**
+
+```js
+module.exports = {
+  parser: "@babel/eslint-parser",
+  parserOptions: {
+    requireConfigFile: false,
+    babelOptions: {
+      babelrc: false,
+      configFile: false,
+    },
+  },
+};
+```
+
 - `sourceType` can be set to `"module"`(default) or `"script"` if your code isn't using ECMAScript modules.
 <!-- TODO(Babel 8): Remove this -->
 - `allowImportExportEverywhere` (default `false`) can be set to `true` to allow import and export declarations to appear anywhere a statement is allowed if your build environment supports that. Otherwise import and export declarations can only appear at a program's top level.

